### DISCUSSION TITLE
detect packet lost for playing on ffmpeg

### DISCUSSION THREAD

**Discussion by amirvalhalla** [bluenviron/gortsplib#229] [Questions]
Hi 

version of package:
v2.1.1

issue:  
I'm trying to connect to RTSP protocol of Axis camera on H264 codec , I will connect and will get data but when I will play it on my player which made it By FFMPEG , the video doesn't play well (seems  I lost some Pframes). so is there a way to detect packet lost to prevent it ?

Tip:
I've used OnWarning callback function and I've packet lost

**Comment by aler9** [comment]
Hello, loss of packet happens when the UDP transport protocol is in use and when the network link between the camera and the player doesn't have a bandwidth wide enough to handle the stream.

You can either

* switch to the TCP transport protocol, that automatically re-transmits lost packets
* update the link between the camera and the player

There's a mechanism for reducing packet losses with UDP called NACK (negative acknowledges) but it has not been implemented in this library due to lack of support from major players (FFmpeg, GStreamer, VLC).
